### ISSUE TITLE
Bug: lualine should not override user options unless explicitly requested

### ISSUE THREAD

**Issue by ZyX-II** [nvim-lualine/lualine.nvim#921]
### Self Checks

- [x] I'm using the latest lualine. (fffbcb829288c3ca366f17cdb8d46aefb5c1d6f3)
- [x] I didn't find the issue in exsisting issues or prs.
       Specifically, #395 does mention the issue regarding `showtabline`, but it contains more generic feature request while this issue is more limited in scope; and existence of #670 suggests that current behaviour of ignoring `laststatus` should be classified as a bug.

### How to reproduce the problem

1. Run neovim with `-u exampleConfig.vim` (replace paths in first line with appropriate ones).
2. Type `:set showtabline? laststatus?`.

### Expected behaviour

Neovim reports both options as being equal to 1.

### Actual behaviour

They are both set to 2.

### Minimal config to reproduce the issue

```
set runtimepath=C:/tools/neovim/nvim-win64/share/nvim/runtime,P:/ath/to/lualine
set laststatus=1
set showtabline=1
lua << EOF
require('lualine').setup({
    tabline = { lualine_a = { 'tabs' } },
})
EOF
```

### Additional information

Workaround using OptionSet from #395 used as-is not good idea because it prevents setting this option to anything else. It is obviously possible to create workaround for this issue of the workaround, but having options to configure options is something strange. Also just setting options after setup call for some reason works in the minimal config above, but not in my actual config.

**Comment by shadmansaleh** [comment]
I disagree with this. Most people using vim/nvim aren't aware of these options or what they do.

And people are usually looking for statuslline & tabline to be always visible. For example, if you're using your tabline to show your gitstatus it doesn't make sense for it only to be visible when there is more than 1 tab. That makes sense when you're using tabline to display only tabs like the builtin tabline . Same goes for statusline I don't think anyone wants their statusline to disappear when they have only one window open.

We are already taking over statusline & tabline options. I don't think there's anything wrong with us taking over a few more related options to provide better default behavior.

As for #395 I do intend to fix it properly. My idea is to make tabline disappear when the components in tabline all return empty. This lets the components decide when they want to show themselves. And the users can use `cond` option to provide their own logic when they want tabline to be shown, instead of relying on just the number of tabs. winbar already works like this. I just didn't get time to do the same for tabline yet.

**Comment by ZyX-I** [comment]
> I disagree with this. Most people using vim/nvim aren't aware of these options or what they do.
>
> And people are usually looking for statuslline & tabline to be always visible. For example, if you're using your tabline to show your gitstatus it doesn't make sense for it only to be visible when there is more than 1 tab. That makes sense when you're using tabline to display only tabs like the builtin tabline . Same goes for statusline I don't think anyone wants their statusline to disappear when they have only one window open.
>
> We are already taking over statusline & tabline options. I don't think there's anything wrong with us taking over a few more related options to provide better default behavior.
> 
> As for https://github.com/nvim-lualine/lualine.nvim/issues/395 I do intend to fix it properly. My idea is to make tabline disappear when the components in tabline all return empty. This lets the components decide when they want to show themselves. And the users can use cond option to provide their own logic when they want tabline to be shown, instead of relying on just the number of tabs. winbar already works like this. I just didn't get time to do the same for tabline yet.

I would actually did not care if I had alternative. `laststatus` part got noticed only while searching for similar issues.

> One more thing is this your account https://github.com/ZyX-I ?

Yes, I created second to use it from work while first is only used from home. Not any real kind of active on either these days though…

**Comment by giodueck** [comment]
This can be closed since https://github.com/nvim-lualine/lualine.nvim/pull/1013 is merged, right? Just coming back to see this is now solved.

**Comment by mostafatouny** [comment]
Here is a workaround which forces `laststatus = 0`
```
vim.api.nvim_create_autocmd("VimEnter", {
  callback = function()
    vim.opt.laststatus = 0
  end,
})
```
